app/controllers/registrations_controller.rb

```diff
@@ -1,6 +1,38 @@
 class RegistrationsController < Devise::RegistrationsController
+  def update
+    update_params = account_update_params
+    # required for settings form to submit when password is left blank
+    if update_params[:password].blank?
+      [:password,:password_confirmation,:current_password].collect{|p| update_params.delete(p) }
+    end
+    if update_params[:password].present? && update_params[:current_password].blank?
+      [:current_password].collect{|p| update_params.delete(p) }
+    end
+
+    self.resource = resource_class.to_adapter.get!(send(:"current_#{resource_name}").to_key)
+    prev_unconfirmed_email = resource.unconfirmed_email if resource.respond_to?(:unconfirmed_email)
+
+    if update_resource(resource, update_params)
+      yield resource if block_given?
+      if is_flashing_format?
+        flash_key = update_needs_confirmation?(resource, prev_unconfirmed_email) ?
```

Comment:
Line is too long. [82/80]<br>Avoid multi-line ?: (the ternary operator); use `if`/`unless` instead.